inst/tutorials/tidyverse_ws/tidyverse_ws.Rmd

```diff
@@ -0,0 +1,1060 @@
+dat
+#arrange(dat, Mean_CH4)
+```
+
+To sort rows in descending order, use `arrange(<data_frame>, desc(<x>))`. 
+In the box below, arrange `dat` in descending order of `Mean_O2`:
+
+```{r arrange_descending, exercise = TRUE, exercise.setup = "setup_dat"}
+arrange(<code here>)
+```
+
+```{r arrange_descending-solution}
+arrange(dat, desc(Mean_O2))
+```
+
+
+### `mutate()`
+
+Mutate is one of the most useful verbs in the dplyr package.
+Use `mutate(data_frame, y = function(x))` to apply a transformation to some column `x` and assign the output to the column name `y`. 
+The transformation can be a function (e.g., `log(x)`) or a mathematical operation. 
+As an example, run the code below to divide `Depth` by 1000 (converting its units from metres to kilometers) and assign the output to a new column, `Depth_km`.
+
+```{r mutate, exercise = TRUE, exercise.setup = "setup_dat"}
+mutate(dat, Depth_km = Depth/1000)
+```
+
+We can also mutate multiple columns. In the box below, try to use `mutate` to create a new column (with any name of your choice) containing the sum of `Mean_O2` and `Mean_CH4`:
+
+```{r mutate_sum, exercise = TRUE, exercise.setup = "setup_dat"}
+mutate(dat, <new column name> = <sum of old columns>)
+```
+
+```{r mutate_sum-solution}
+mutate(dat, O2_plus_CH4 = Mean_O2 + Mean_CH4)
+```
+
+#### Exercise: Rename and Mutate
+
+Using the data frame `dat`:
+
+1. Select the `Date`, `Depth` and `Mean_O2` variables
+2. Rename the `Mean_O2` column to `O2`
+3. Keep August observations where oxygen data are non-missing using `filter()`, `month()`, and `!is.na()`
+4. Transform `O2` from micromoles/L to micrograms/L using `mutate()` (multiply by 32)
+
+At each step, save the output into a new data frame with a name of your choice.
+Use the new data frame as an argument for the next step.
+The final data frame should have sixteen rows. 
+
+```{r exercise_rename_mutate, exercise = TRUE, exercise.setup = "setup_dat"}
+
+
+
+```
+
+```{r exercise_rename_mutate-solution}
+selected_data <- select(dat, Date, Depth, Mean_O2)
+renamed_data <- rename(selected_data, O2 = Mean_O2)
+filtered_data <- filter(renamed_data, month(Date) == 8 & !is.na(O2))
+mutate(filtered_data, O2 = O2 * 32)
+```
+
+### Piping with `%>%`
+
+In the previous exercise, we had to perform quite a large series of data wrangling tasks.
+First, we selected some columns; then, we renamed them; then we filtered them; and finally we mutated one of the columns.
+The most natural way to think about this exercise is as a *chain* (or sequence) of functions: each dplyr verb wrangles the input data frame into a new (output) data frame, which is then provided as the input to the following verb.
+Metaphorically, the output of the first function is *piped into* the following function.
+
+The tidyverse package magrittr introduced the `%>%` (pipe) operator, which uses this metaphor to make code easier to read:
+
+`f(x) %>% g(y)` is the same as `g(f(x),y)`
```

Comment:
Nice! This is a very succinct way of explaining piping, and this exact line is what made it make sense to me when I was first learning about the pipe operator. Well done!